### PR TITLE
fix(middleware): Catch errors when loading a module (#3572)

### PR DESCRIPTION
**Pull request by jehon**
Fix karma-runner/karma#3572

### CODE REVIEW

File: lib/middleware/karma.js [hunk 1]
```diff
@@ -193,7 +193,11 @@ function createKarmaMiddleware (
             } else {
               const scriptType = (SCRIPT_TYPE[fileType] || 'text/javascript')
               const crossOriginAttribute = includeCrossOriginAttribute ? 'crossorigin="anonymous"' : ''
-              scriptTags.push(`<script type="${scriptType}" src="${filePath}" ${crossOriginAttribute}></script>`)
+              if (fileType === 'module') {
```
Comment (johnjbarton) [inline lib/middleware/karma.js:196]: Why the `if`.  Is there a reason to to have the `onerror` for both paths?

Comment (jehon) [reply lib/middleware/karma.js:196]: I did that that way to reduce the impact... I wonder if the error would show up twice without the if

Comment (jehon) [reply lib/middleware/karma.js:196]: I did make the test. Without the if, there are tests that does not pass anymore (5 in the first phase). So the "if" is necessary to reduce the impact of the change.

**Comment by jehon** [comment]
@johnjbarton Sorry to interrupt you, but could you have an (updated) opinion on this PR?
The basis is that, without this PR and actually, karma report success when there is a syntax error in a es6 test.

**Comment by jehon** [comment]
Could someone review this PR?
If some work need to be done, could it be said?
Thanks

**Comment by johnjbarton** [comment]
Sorry for the long delay. I want to investigate the possibility of correcting any tests that fail with the error on both paths.

**Comment by jehon** [comment]
I did look into the failings tests, nothing important. It's just that it test the string, and that in need to be adapted. I can push it here if wanted.

But the problem is that the output would change for errors in normal scripts (here, I put a call to "nothing()":

```
Uncaught ReferenceError: nothing is not defined
ReferenceError: nothing is not defined
   at tests/failing-test.js:2:1
```

Instead of
```
ReferenceError: nothing is not defined
   at tests/failing-test.js:2:1
```

There is no difference with the "if".

**Comment by jehon** [comment]
Followup in https://github.com/karma-runner/karma/pull/3585

**Comment by jehon** [comment]
Followup in https://github.com/karma-runner/karma/pull/3605

**Pull request closed without merging**